src/Vault.sol

```diff
@@ -157,7 +157,7 @@ contract Vault is IVault, VaultToken, Ownable {
     }
 
     /// @inheritdoc IVault
-    function collectFee(Currency currency, uint256 amount, address recipient) external {
+    function collectFee(Currency currency, uint256 amount, address recipient) external onlyRegisteredApp {
```

Comment:
might be good to add a test to check if theres revert if called by non registered app

Reply:
yeah, there is already that check. can see the test case i updated below